### PR TITLE
Expose Color's `to_linear()` and `to_srgb()` to scripting

### PR DESCRIPTION
**Pull request by Calinou**
This also adds unit tests for `to_linear()` and `to_srgb()`.

See https://github.com/godotengine/godot-proposals/discussions/4246#discussioncomment-2410357.

### CODE REVIEW

File: tests/core/math/test_color.h [hunk 1]
```diff
@@ -144,6 +144,24 @@ TEST_CASE("[Color] Conversion methods") {
 			"The string representation should match the expected value.");
 }
 
+TEST_CASE("[Color] Linear <-> sRGB conversion") {
+	const Color color = Color(0.35, 0.5, 0.6, 0.7);
+	const Color color_linear = color.to_linear();
+	const Color color_srgb = color.to_srgb();
+	CHECK_MESSAGE(
+			color_linear.is_equal_approx(Color(0.1005, 0.214, 0.3185, 0.7)),
```
Comment (Calinou) [inline tests/core/math/test_color.h:152]: This test fails and I have no idea why – it's matching the exact value when `color_linear` is printed. Same for `color_srgb` below.

The bottom two tests are successful though.

Comment (Calinou) [reply tests/core/math/test_color.h:152]: I wonder if this something we should fix in the sRGB/linear conversion functions, or just use a custom epsilon for this check.

Comment (akien-mga) [reply tests/core/math/test_color.h:152]: Is it actually losing precision? Seems to me like the issue is that you are using a reference value which has been rounded to 0.003 and that's lossy/only for pretty printing.

Comment (bruvzg) [reply tests/core/math/test_color.h:152]: Is it loosing precision? I'm getting the same value with the calculator, e.g., `R = 0,1004815099`. I guess it's `print` what's loosing precision.

Comment (Calinou) [reply tests/core/math/test_color.h:152]: Indeed, the print was only printing 4 decimals (both in GDScript and C++). I'm not sure if it should be rounding this much.

**Review by clayjohn — APPROVED** [review]
I don't know anything about tests, but I support exposing these functions.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**